gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -0,0 +1,155 @@
+  // and add them to the register definitions.
+  for (unsigned i = 0;
+       i < MRI.getRegClass(llvm::X86::GR64_NOREXRegClassID).getNumRegs(); ++i) {
+    llvm::StringRef reg_name =
+        MRI.getName(MRI.getRegClass(llvm::X86::GR64_NOREXRegClassID).getRegister(i));
+    register_defs_lines += std::string(kRegDefPrefix) + std::string(reg_name) +
+                           " " + std::to_string(kInitialRegVal) + "\n";
+  }
+  for (unsigned i = 0; i < MRI.getRegClass(llvm::X86::VR128RegClassID).getNumRegs();
+       ++i) {
+    llvm::StringRef reg_name =
+        MRI.getName(MRI.getRegClass(llvm::X86::VR128RegClassID).getRegister(i));
+    register_defs_lines += std::string(kRegDefPrefix) + std::string(reg_name) +
+                           " " + std::to_string(kInitialRegVal) + "\n";
+  }
+
+  gematria::X86Canonicalizer canonicalizer(&llvm_support->target_machine());
+  gematria::BHiveImporter bhive_importer(&canonicalizer);
+
+  std::ifstream bhive_csv_file(bhive_filename);
+  for (std::string line; std::getline(bhive_csv_file, line);) {
+    auto comma_index = line.find(',');
+    if (comma_index == std::string::npos) {
+      std::cerr << "Invalid CSV file: no comma in line '" << line << "'\n";
+      return 2;
+    }
+
+    std::string_view hex = std::string_view(line).substr(0, comma_index);
+    // For each line, find the accessed addresses & disassemble instructions.
+    auto bytes = gematria::ParseHexString(hex);
+    if (!bytes.has_value()) {
+      std::cerr << "could not parse: " << hex << "\n";
+      return 3;
+    }
+
+    // This will only get the first segfault address.
+    auto addrs = gematria::FindAccessedAddrs(*bytes);
+    auto proto = bhive_importer.BasicBlockProtoFromMachineCode(*bytes);
+
+    // Check for errors.
+    if (!proto.ok()) {
+      std::cerr << "Failed to disassemble block '" << hex << ": "
+                << proto.status() << "\n";
+      continue;
+    }
+    if (!addrs.ok()) {
+      std::cerr << "Failed to find addresses for block '" << hex
+                << "': " << addrs.status() << "\n";
+      std::cerr << "Block disassembly:\n";
+      for (const auto& instr : proto->machine_instructions()) {
+        std::cerr << "\t" << instr.assembly() << "\n";
+      }
+      continue;
+    }
+
+    // Create output file path.
+    std::string output_file_path =
+        output_dir + "/" + std::to_string(file_counter) + ".test";
+
+    // Open output file for writing.
+    std::ofstream output_file(output_file_path);
+    if (!output_file.is_open()) {
+      std::cerr << "Failed to open output file: " << output_file_path << "\n";
+      return 4;
+    }
+
+    // Write the register definition lines into the output file.
+    output_file << register_defs_lines;
+
+    // Multiple mappings can point to the same definition.
+    if (addrs->accessed_blocks.size() > 0) {
+      output_file << kMemDefPrefix << kMemNamePrefix << " " << addrs->block_size
+                  << " " << kInitialMemVal << "\n";
```

Comment:
`kInitialMemVal` is a hex value, so we should convert the integer to a hex value here rather than just printing the decimal number.